app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.wallpapers
+
+import android.view.View
+import mozilla.components.support.ktx.android.content.getColorFromAttr
+import org.mozilla.fenix.utils.Settings
+
+/**
+ * Provides access to available wallpapers and manages their states.
+ */
+class WallpaperManager(private val settings: Settings) {
+
+    var currentWallpaper: Wallpaper = getCurrentWallpaperFromSettings()
+        set(value) {
+            settings.currentWallpaper = value.name
+            field = value
+        }
+
+    /**
+     * Apply the [newWallpaper] into the [wallpaperContainer] and update the [currentWallpaper].
+     */
+    fun updateWallpaper(wallpaperContainer: View, newWallpaper: Wallpaper) {
+        if (newWallpaper == Wallpaper.NONE) {
+            val context = wallpaperContainer.context
+            wallpaperContainer.setBackgroundColor(context.getColorFromAttr(newWallpaper.drawable))
+        } else {
+            wallpaperContainer.setBackgroundResource(newWallpaper.drawable)
+        }
+        currentWallpaper = newWallpaper
+    }
+
+    /**
+     * Apply the next wallpaper into the [wallpaperContainer] and update the [currentWallpaper].
+     */
+    fun switchToNextWallpaper(wallpaperContainer: View) {
+        val values = Wallpaper.values()
+        val index = values.indexOf(currentWallpaper) + 1
+
+        val wallpaper = if (index >= values.size) {
+            values.first()
+        } else {
+            values[index]
+        }
+        updateWallpaper(wallpaperContainer, wallpaper)
+    }
+
+    private fun getCurrentWallpaperFromSettings(): Wallpaper {
+        val currentWallpaper = settings.currentWallpaper
+        return if (currentWallpaper.isEmpty()) {
+            Wallpaper.NONE
+        } else {
+            Wallpaper.valueOf(currentWallpaper)
+        }
+    }
```

Comment:
The logic in this function could be moved into the getter for the `currentWallpaper` property to keep a consistent style with the setter.

Reply:
We are doing this to just read from disk (shared pref)  ones, when initializing `currentWallpaper`, then the current value will be stored in memory [instead of reading from disk every time the property is accessed (get)](https://kotlinlang.org/docs/properties.html#backing-fields)

Reply:
Gotcha, that makes sense